src/charm.py

```diff
@@ -1,88 +1,114 @@
+        self.framework.observe(self.on.login_ui_pebble_ready, self._on_login_ui_pebble_ready)
         self.framework.observe(self.on.config_changed, self._on_config_changed)
         self.framework.observe(self.ingress.on.ready, self._on_ingress_ready)
+        self.framework.observe(self.ingress.on.revoked, self._on_ingress_revoked)
+
+    def _on_login_ui_pebble_ready(self, event: WorkloadEvent) -> None:
+        """Define and start a workload using the Pebble API."""
+        self._handle_status_update_config(event)
+
+    def _on_config_changed(self, event: ConfigChangedEvent) -> None:
+        """Handle changed configuration."""
+        self._handle_status_update_config(event)
+
+    def _handle_status_update_config(self, event: HookEvent) -> None:
+        if not self._container.can_connect():
+            event.defer()
+            logger.info(
+                """Cannot connect to Login_UI container.
+ Deferring the event."""
+            )
+            self.unit.status = WaitingStatus("Waiting to connect to Login_UI container")
+            return
+
+        self.unit.status = MaintenanceStatus("Configuration in progress")
+
+        self._container.add_layer(self._container_name, self._login_ui_layer, combine=True)
+        logger.info("Pebble plan updated with new configuration, replanning")
+        try:
+            self._container.replan()
+        except ChangeError as err:
+            logger.error(str(err))
+            self.unit.status = BlockedStatus("Failed to replan, please consult the logs")
+            return
 
-    @property
-    def _container(self):
-        return self.unit.get_container(self._container_name)
-
-    def _on_kratos_ui_pebble_ready(self, event):
-        """Define and start a workload using the Pebble API.
-        """
-        self._container.add_layer("kratos_ui", self._pebble_layer, combine=True)
-        self._container.replan()
         self.unit.status = ActiveStatus()
 
-    def _on_config_changed(self, event):
-        """Handle changed configuration.
-        """
-        # Fetch the new config value
-        log_level = self.model.config["log-level"].lower()
+    def _on_ingress_ready(self, event: IngressPerAppReadyEvent) -> None:
+        if self.unit.is_leader():
+            logger.info("This app's public ingress URL: %s", event.url)
 
-        self.unit.status = MaintenanceStatus("Applying configuration")
-        self._on_kratos_ui_pebble_ready(None)
+    def _on_ingress_revoked(self, event: IngressPerAppRevokedEvent) -> None:
+        if self.unit.is_leader():
+            logger.info("This app no longer has ingress")
 
     @property
-    def _pebble_layer(self):
-        """Return a dictionary representing a Pebble layer."""
-        return {
-            "summary": "kratos ui layer",
-            "description": "pebble config layer for kratos_ui",
+    def _login_ui_layer(self) -> Layer:
+        # Define an initial Pebble layer configuration
```

Comment:
This is not only an initial layer, it can change due to config
```suggestion
        # Define Pebble layer configuration
```

Reply:
done